tests/test_sdsc.py

```diff
@@ -59,6 +59,14 @@ def test_highlighter():
     assert xml == "highlight <highlight>these two</highlight> words"
 
 
+def test_contextpatternlocations():
+    """checks the contextpatternlocations function"""
+    assert sdsc.contextpatternlocations(1, [1]) == [1]
+    assert sdsc.contextpatternlocations(1, [1, 2]) == [1, 2]
```

Comment:
IMO that's the caller's job. If it only accepts three different kinds of values, it shouldn't be a list as argument.